### ISSUE TITLE
Argument lists for module calls through publish.publish does not work any more

### ISSUE THREAD

**Issue by mikn** [saltstack/salt#18317]
According to the documentation here: http://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.publish.html#salt.modules.publish.publish
You're supposed to be able to send in a comma-separated list in a string that will be matched against the positional arguments that the module call you're making. This no longer works. I traced it to this commit: https://github.com/saltstack/salt/commit/36fa800821173a4bb1efc950d92b58602c232304
And this issue: #13294
It seems that the argument parser was switched out to a YAML-like parsing style. This lead me to trying a few other types of syntax including YAML style "[item1, item2]" and Python style "['item1', 'item2']" without success. I will locally revert the publish module since it seems that this is a larger issue than just a simple patch.

**Comment by luxi2001** [comment]
I am  also running into this issue. @mikn, Can you provide the details of your workaround?

**Comment by mikn** [comment]
@luxi2001 The problem lies on this row: https://github.com/saltstack/salt/blob/2014.7/salt/modules/publish.py#L58
It breaks if you want to send more than one argument in. So, what you need to do is something like this (untested code) instead of that single line:
```python
    parsed_args = salt.utils.args.yamlify_arg(arg)
    if isinstance(parsed_args, list):
        arg = parsed_args
    else:
        arg = [parsed_args]
```
The syntax for a yaml list is slightly different, too. You need to do `[value, value, value]` and not just `value, value, value` as it was before. If the values contain commas, you must quote them too, like: `['value, hi', value, 'value, test']` to make it work properly.

If you're using RAET, you must change that line in raet_publish.py and not publish.py :)

**Comment by jfindlay** [comment]
@mikn, is [this line](https://github.com/saltstack/salt/blob/2014.7/salt/modules/publish.py#L238) also affected?  It came in on the [same commit](#13370).

**Comment by jfindlay** [comment]
@mikn, I hope it's fine with you that I've fixed this according to your suggestions.

**Comment by mikn** [comment]
Oh, of course! Today was a very busy day and I have not validated it
completely, but if you feel like it solved your problem and you've tested
the four main scenarios (no arguments, one argument, list of arguments and
keyword arguments), all is good!

**Comment by ryan-lane** [comment]
Will this be backported to 2014.7?

**Comment by thatch45** [comment]
it should be.
@jfindlay please backport if needed

**Comment by jfindlay** [comment]
Here I am getting bugged at fixes carelessly getting set to be pulled into develop when they apply previously and then I do it myself. :-(

**Comment by jfindlay** [comment]
So, there are two parts to this fix:
- #20992
- #21047

Both were originally merged into 2014.7 and have both since made it through 2015.2 into develop.

**Comment by apergos** [comment]
so the next question then is when a new release of 2014.7 will be available with this fix, for those of us who steal the salt ppa packages.

**Comment by rallytime** [comment]
@apergos This fix will be available in 2014.7.3, as I don't think this made it into 2014.7.2. However, I am not sure when that release is planned to come out yet. It should be within the next couple of weeks or so.

**Comment by apergos** [comment]
we can hang on for two weeks (hope the packages come real soon after that). thanks!
